src/runner/Run_semgrep.ml

```diff
@@ -624,6 +652,7 @@ let targets_of_config (config : Runner_config.t)
         | Target_file target_file ->
             File.read_file target_file |> In.targets_of_string
       in
+      let targets = hack_targets_hash_consing targets in
```

Comment:
My only concern is that this could cause problems if we transform the targets later. There are two places we mess with the targets list: extract mode, and now secrets. I believe for both of these we're just adding new targets, but it doesn't seem implausible that one day we'd want to remove a rule from a target list for a specific target